potential fix for CI

kostrykin opened the pull request:
The requirement is that the CI deploys tools even if planemo lint raises an error during merge.

The current configuration of the CI sets`--fail_level` passed to `planemo shed_lint` to `error`:
https://github.com/BMCV/galaxy-image-analysis/actions/runs/6811812716/job/18523060276#step:7:355
To my understanding, this makes the linter fail only if it produces an error (but not if it produces warnings).
https://help.galaxyproject.org/t/make-a-repository-installable/8156/6

What we want instead, is, that the linter never fails during merge, not even if an error is encountered! However, only the options `warn` and `error` are allowed:
```
$ planemo shed_lint --help |grep fail_level
  --fail_level [warn|error]
```

Thus, the required behavior is out of scope of `fail_level`. It can only be achieved by skipping linting during merge. This should be implemented by this PR (hopefully, never digged myself into GitHub actions before).

xref #82 #71 

---

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/BMCV/galaxy-image-analysis/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the galaxy-image-analysis repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [x] - This PR does something else (explain below)
* [ ] - Tools added/updated by this PR (if any) comply with the [Naming and Annotation Conventions for Tools in the Image Community in Galaxy](https://github.com/elixir-europe/biohackathon-projects-2023/blob/main/16/conventions.md) (or please explain why they do not)

bernt-matthias commented:
This is not needed IMO. For deployment lint runs with `--fail-level error` anyways. 

Having `--fail-level warn`  during PRs seems a nice feature since it usually leads to higher quality tools.

kostrykin commented:
> Having `--fail-level warn` during PRs seems a nice feature since it usually leads to higher quality tools.

Thanks, you are right, I changed this by accident. I fixed that in 69065ea.

bernt-matthias commented:
Sorry, this is not what I meant .. `warn` is even stricter that `fail`, i.e. if it fails with `error` it will fail with `warn` as well. 

Give me a moment. 

Ultimately biii needs to be added [here](https://github.com/galaxyproject/galaxy/blob/f91693dc6e9a252502cdcc574e0c07c3d234cb5d/lib/galaxy/tool_util/xsd/galaxy.xsd#L7405C10-L7405C10) .. which will effect only with the next galaxy release.

kostrykin commented:
> Sorry, this is not what I meant .. warn is even stricter that fail, i.e. if it fails with error it will fail with warn as well.

Yes, this is why I have changed it from `error` to `warn` for pull requests (because it's stricter). For merges, planemo lint would be skipped completely due to f976b91.

bernt-matthias commented:
https://github.com/BMCV/galaxy-image-analysis/pull/84
